operations/sysinfo.go

```diff
@@ -145,30 +149,49 @@ func processTree() cli.Command {
 //
 ///////////////////////////////////////////////////////////////////////////
 
-func getLogger(fn string) (grip.Journaler, error) {
+type closer func()
+
+func getLogger(fn string) (grip.Journaler, closer, error) {
+	closer := func() {}
 	logger := grip.NewJournaler("curator.stats")
 	sender := logger.GetSender()
 	lvl := sender.Level()
 	lvl.Threshold = level.Debug
 
 	if err := sender.SetLevel(lvl); err != nil {
-		return nil, errors.Wrap(err, "problem setting logging threshold")
+		return nil, closer, errors.Wrap(err, "problem setting logging threshold")
 	}
 
 	if fn != "" {
-		sender, err := send.MakeJSONFileLogger(fn)
-		if err != nil {
-			return nil, errors.Wrap(err, "problem building logger")
+		var sender send.Sender
+		var err error
+
+		if _, err = os.Stat(fn); os.IsNotExist(err) {
+			sender, err = send.MakeJSONFileLogger(fn)
+			if err != nil {
+				return nil, closer, errors.Wrap(err, "problem building logger")
+			}
```

Comment:
You need to set a closer for this case.

Reply:
I don't think so, the sender provides its own closer, which handles this case (and anyway, we've never explicitly closed that in this case.